src/plugins/workspace/public/components/workspace_detail/data_source_connection_table.tsx

```diff
@@ -152,19 +165,20 @@ export const DataSourceConnectionTable = ({
         ]
       : []),
     {
-      width: '25%',
+      width: '20%',
       field: 'name',
       name: i18n.translate('workspace.detail.dataSources.table.title', {
         defaultMessage: 'Title',
       }),
       truncateText: true,
       render: (name: string, record) => {
         const origin = window.location.origin;
+        const basePath = http.basePath.serverBasePath;
         let url: string;
         if (record.connectionType === DataSourceConnectionType.OpenSearchConnection) {
-          url = `${origin}/app/dataSources/${record.id}`;
+          url = `${origin}${basePath}/app/dataSources/${record.id}`;
```

Comment:
Can we use `basePath.prepend(`/app/dataSources/${record.id}`, { withoutClientBasePath: true })`?

Reply:
`basePath` is a string and there's no `prepend` in `basePath`

Reply:
right, please use http.basePath to get the final path. We should avoid string concatenation when generating the path.